vite.config.ts

```diff
@@ -27,6 +27,11 @@ export default defineConfig(({ mode }) => {
         define: {
             // Preserve process.env access for compatibility
             "process.env.NODE_ENV": JSON.stringify(env.NODE_ENV || mode),
+            // preserve process.env.VITE_ variables for backwards compatibility with CJS
+            "process.env.VITE_PRESENTATION_VARIANT": JSON.stringify(env.VITE_PRESENTATION_VARIANT),
+            "process.env.VITE_PRESENTATION_TITLE": JSON.stringify(env.VITE_PRESENTATION_TITLE),
+            "process.env.VITE_PRESENTATION_TYPE": JSON.stringify(env.VITE_PRESENTATION_TYPE),
+            "process.env.VITE_CYPRESS": JSON.stringify(env.VITE_CYPRESS),
```

Comment:
This also caused issues for us in CPR. Just as a note, what we did there was use `vite-node` to run the scripts so that the `import.meta` references keep working.

That’s not to say this approach is wrong, moving everything to environment variables is reasonable.